proof-systems/src/darlin/poly-commit/src/ipa_pc/data_structures.rs

```diff
@@ -726,6 +726,30 @@ impl<G: IPACurve> SemanticallyValid for SuccinctCheckPolynomial<G> {
     }
 }
 
+/// Succinct check polynomial tagged with a label
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct LabeledSuccinctCheckPolynomial<'a, G: IPACurve> {
```

Comment:
Not a big fan of it honestly. I would've preferred re-using the LabeledPolynomial struct (you can compute the coefficients of the Bullet poly from the xi_s, the function should return a Polynomial instance that you can use to construct a LabeledPolynomial

Reply:
I thought it would have been more inefficient, as computing the actual polynomial is much more computationally expensive than employing the `SuccinctCheckPolynomial` representation. 

For instance, I needed to add a label to `SuccinctCheckPolynomial` for the `open_reduction_polynomials` function, which computes the polynomial from the succinct representation inside the function, scaling also the coefficient by a squeezed challenge during the [construction](https://github.com/HorizenOfficial/ginger-lib/blob/add_poly-commit_verifier_gadget/proof-systems/src/darlin/poly-commit/src/ipa_pc/mod.rs#L177). If I would require the function to take a `LabeledPolynomial` rather than `LabeledSuccinctCheckPolynomial`, then the caller should have computed the bullet polynomial, and the `open_reduction_polynomial` would have needed to iterate over all the coefficients to scale them by the squeezed challenge, which is twice more costly than the current implementation. 

In general, if the user of the `SuccinctCheckPolynomial` just needs a label for the succinct representation of the bullet polynomial, it would be sometimes much more inefficient to force it to compute the polynomial coefficients explicitly. That is the reason why I decided to

Reply:
Not my favourite solution but ok. Double check also with @lgiussan that this construction is fine when coming to dlog accumulation in marlin

Reply:
Eventually I realized that there was no need to call domain extended IPA in `succinct_verify_accumulated_items` of `DLogItemAccumulator`, as the function only deals with bullet polynomials, which always have only 1 segment. By calling the simple IPA poly-commit, there is no longer the need to sort commitments in the `open_reduction_polynomials` function, in turn removing the need for the `SuccinctCheckPolynomial`s provided to such function to have a label. In conclusion, the `LabeledSuccinctCheckPolynomial` struct seems to be no longer needed.